In [16]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

from scipy.stats import norm

In [2]:
df = pd.read_csv("Sachin_ODI.csv")

In [3]:
df.head()

runs  NotOut mins  bf  fours  sixes      sr  Inns           Opp    Ground  \
0    13       0   30  15      3      0   86.66     1   New Zealand    Napier   
1    37       0   75  51      3      1   72.54     2  South Africa  Hamilton   
2    47       0   65  40      7      0  117.50     2     Australia   Dunedin   
3    48       0   37  30      9      1  160.00     2    Bangladesh   Sharjah   
4     4       0   13   9      1      0   44.44     2      Pakistan   Sharjah   

         Date        Winner    Won  century  
0  1995-02-16   New Zealand  False    False  
1  1995-02-18  South Africa  False    False  
2  1995-02-22         India   True    False  
3  1995-04-05         India   True    False  
4  1995-04-07      Pakistan  False    False

In [4]:
mu = df["runs"].mean()
mu

43.24166666666667

In [5]:
sigma = df["runs"].std()
sigma

42.18245206052441

In [6]:
df.groupby(["Inns"])["runs"].mean()

Inns
1    46.670588
2    40.173684
Name: runs, dtype: float64

In [7]:
df.groupby(["Won"])["runs"].mean()

Won
False    35.130682
True     51.000000
Name: runs, dtype: float64

In [9]:
mu

43.24166666666667

In [10]:
first_innings_runs = df[df["Inns"]==1][["runs"]]

In [13]:
len(first_innings_runs)

170

In [11]:
data = first_innings_runs

In [12]:
data.mean()

runs    46.670588
dtype: float64

In [15]:
z_stat = (data.mean() - mu) / (sigma / np.sqrt(len(data)))
z_stat

runs    1.059864
dtype: float64

In [18]:
p_value = 1 - norm.cdf(z_stat)
p_value

array([0.14460321])

In [19]:
alpha = 0.05
if p_value < alpha:
    print("Reject Null Hypotheis")
else:
    print("Stick to the null hypothesis")

Stick to the null hypothesis


Scoring pattern for matches won

H0: average runs in matches won = 43.2
H0: average runs in matches won > 43.2

In [21]:
victory_runs = df[df["Won"]==True][["runs"]]
data = victory_runs

In [23]:
data.mean()

runs    51.0
dtype: float64

In [24]:
len(data)

184

In [22]:
z_stat = (data.mean() - mu) / (sigma / np.sqrt(len(data)))
z_stat

runs    2.494856
dtype: float64

In [25]:
p_value = 1 - norm.cdf(z_stat)
p_value

array([0.00630041])

In [26]:
alpha = 0.05
if p_value < alpha:
    print("Reject Null Hypotheis")
else:
    print("Stick to the null hypothesis")

Reject Null Hypotheis
